platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -39,6 +39,34 @@ For Java, Groovy, Kotlin, and Android compatibility, see the [full compatibility
 
 ## New features and usability improvements
 
+### Publishing improvements
+
+#### Introduced `PublishingExtension#getSoftwareComponentFactory` method
+
+This new methods exposes the `SoftwareComponentFactory` service without requiring it to be injected.
+Without this service accessor, it is not possible to create and publish custom software components in build scripts or precompiled script plugins without resorting to more cumbersome means of acquiring the service.
+
+Consider the following example showcasing how to publish a custom component using this new method.
+
+```kotlin
+plugins {
+    id("maven-publish")
+}
+
+val consumableConfiguration = getAConfiguration()
+val myCustomComponent = publishing.softwareComponentFactory.adhoc("myCustomComponent").apply {
+    addVariantsFromConfiguration(consumableConfiguration) {}
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("maven") {
+            from(myCustomComponent)
+        }
+    }
+}
```

Comment:
```suggestion
val consumableConfiguration = configurations.consumable("myConsumable").get()

publishing {
	val factory = softwareComponentFactory.adhoc("myCustomComponent")
	factory.addVariantsFromConfiguration(consumableConfiguration) 

    publications {
        create<MavenPublication>("maven") {
            from(myCustomComponent)
        }
    }
}
```

I think that using a Kotlin `apply` isn't a good idea, as it is less straightforward an example than just accessing the factory within the block (which also minimizes its scope).  I don't see a need for `getAConfiguration()` vs. just showing it and making a fully runnable example.

Reply:
I remove the apply and moved the creation of the component into the publishing block. 

But, for now I don't want to make this a complete runnable example, for the sake of reducing duplication